File: dircmp/src/org/dc/dt/dstats.go
package dt

import (
	"fmt"
)

func (s *Stats) String() string {
	var s1 = "dir"
	var s2 = "file"
	var s3 = "byte"
	if s.Dirs != 1 {
		s1 += "s"
	}
	if s.Files != 1 {
		s2 += "s"
	}
	if s.Bytes != 1 {
		s3 += "s"
	}
	return fmt.Sprintf("%s, %s, %s", FmtInt(s.Dirs, s1), FmtInt(s.Files, s2), FmtInt2(s.Bytes, s3))
}

func (s *Copied) String() string {
	return fmt.Sprintf("copied %s", s.Stats.String())
}

func (s *Deleted) String() string {
	return fmt.Sprintf("deleted %s", s.Stats.String())
}

func (s *Scanned) String() string {
	return fmt.Sprintf("scanned %s", s.Stats.String())
}

func (ds *DStats) String() string {
	return fmt.Sprintf("%s\n", ds.Deleted.String()) +
		fmt.Sprintf("%s\n", ds.Copied.String()) +
		fmt.Sprintf("%s\n", ds.Scanned.String()) +
		fmt.Sprintf("%d errors/failures.\n", ds.Errors)
}

File: scala-test/src/main/scala/DStats.scala
import java.text.NumberFormat
import java.util.Locale

object DStats {
	@volatile var bcopy: Long = 0
	@volatile var bdel: Long = 0
	@volatile var dircnt: Int = 0
	@volatile var filecnt: Int = 0

	def update(): Unit = {
		DirSyncApp.onFX {
			val nf = NumberFormat.getNumberInstance(Locale.US)
			DirSyncApp.lbl2(0).text.update("Bytes copied: " + nf.format(DStats.bcopy))
			DirSyncApp.lbl2(1).text.update("Bytes deleted: " + nf.format(DStats.bdel))
			DirSyncApp.lbl2(2).text.update("Dirs scanned: " + nf.format(DStats.dircnt))
			DirSyncApp.lbl2(3).text.update("Files checked: " + nf.format(DStats.filecnt))
		}
	}
}

